### PR TITLE
[Spark][Sharing] Add streaming support in delta format sharing

### PR DESCRIPTION
**Pull request by linzhou-db**
#### Which Delta project/connector is this regarding?
- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (delta sharing)

## Description
Fourth PR of https://github.com/delta-io/delta/issues/2291: Adds Streaming support for "delta format sharing", and add column mapping test
- DeltaSharingDataSource with streaming query support
- DeltaFormatSharingSource
- DeltaFormatSharingSourceSuite/DeltaSharingDataSourceCMSuite

## How was this patch tested?
Unit Tests

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1108,7 +1108,7 @@ val createTargetClassesDir = taskKey[Unit]("create target classes dir")
 
 // Don't use these groups for any other projects
 lazy val sparkGroup = project
-  .aggregate(spark, contribs, storage, storageS3DynamoDB, iceberg, testDeltaIcebergJar, sharing)
```
Comment (vkorukanti) [inline build.sbt:1111]: what is this change?

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceCMSuite.scala [hunk 2]
```diff
@@ -0,0 +1,977 @@
...
+  DeltaUnsupportedOperationException
+}
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.delta.sharing.DeltaSharingTestSparkUtils
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.streaming.{StreamingQueryException, StreamTest, Trigger}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructType}
+
+class DeltaSharingDataSourceCMSuite
```
Comment (vkorukanti) [inline sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceCMSuite.scala:34]: nit: add docs CM -> column mapping.

Comment (vkorukanti) [inline sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceCMSuite.scala:34]: why do you need another suite with the tests repeating? can we have the CM suite extend the `DeltaFormatSharingSourceSuite` with `beforeAll` handling the default column mapping settings for created tables? Just like how we have suites in `Delta-Spark`.

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+        prepareMockedClientMetadata(deltaTableName, sharedTableName)
+        val profileFile = prepareProfileFile(inputDir)
+        val tablePath = profileFile.getCanonicalPath + s"#share1.default.$sharedTableName"
+
+        withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+          def InsertToDeltaTable(values: String): Unit = {
+            sql(s"INSERT INTO $deltaTableName VALUES $values")
+          }
+
+          def processAllAvailableInStream(): Unit = {
```
Comment (vkorukanti) [inline sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala:208]: do these tests simplified if you `testStream` (or the [streaming test framework](DeltaFormatSharingSourceSuite)) is used?

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala [hunk 4]
```diff
@@ -0,0 +1,866 @@
...
+
+          // The streaming query will fail because deletes detected in version 3. And there are no
+          // options provided to ignore the deletion.
+          val e = intercept[Exception] {
+            processAllAvailableInStream(
+              Map("responseFormat" -> "delta", "startingVersion" -> "0")
+            )
+          }
+          for (msg <- Seq(
+              "Detected a data update",
```
Comment (vkorukanti) [inline sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala:476]: nit: why can't you just use the wild card within the expected string for verifying just once? Seems like this type of check is adding decent amount of lines to the test code.

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceCMSuite.scala (hunk 2 above)
Comment (vkorukanti) [reply sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceCMSuite.scala:34]: also add comments on what specifically is this suite testing?

**Review by vkorukanti — CHANGES_REQUESTED** [review]

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala (hunk 4 above)
Comment (linzhou-db) [reply sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala:483]: will check

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala [hunk 5]
```diff
@@ -49,6 +50,11 @@ class DeltaFormatSharingSourceSuite
 
   import testImplicits._
 
+  override def beforeAll(): Unit = {
```
Comment (vkorukanti) [inline sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala:53]: this seems like something you can have in the common test trait: `DeltaSharingDataSourceDeltaTestUtils `

**Review by vkorukanti — APPROVED** [review]

File: sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala (hunk 5 above)
Comment (linzhou-db) [reply sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala]: good point, moved.

**Pull request closed without merging**